### DISCUSSION TITLE
Renovate not resolving complex Gradle variables

### DISCUSSION THREAD

**Discussion by hendrikpeilke** [renovatebot/renovate#24261] [Suggest an Idea]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.78.6

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

In renovate 29.36.2 nested maven repository definitions using variables were resolved correctly, but in renovate 36.78.6 they don't get read. I've created a minimal reproduction at https://github.com/hendrikpeilke/renovate-36.78.6-gradle-repository-bug.

The problem is using constructs like that in `build.gradle` (or `repos.gradle`):
```
def someNonWorkingMavenRepo = {
    name = "SomeName"
    url "https://some.not.working.repository.com/"
}

repositories {
    maven someNonWorkingMavenRepo 
}
```
While this construct ist working:
```
def someWorkingMavenRepoUrl = "https://some.working.repository.com"

repositories {
    maven {
	  url someWorkingMavenRepoUrl
	}
}
```

Unfortunately, we can not switch to the second variant, because of reusing this variable in multiple places and some if statements inside the object, which I left out of the example, to not make it more complicated, than it has to be.

The minimal reproduction can be best analyzed in debug mode on onboarding. Thus no renovate-config is needed.

### Relevant debug logs

As you can see, only the working repository of the minimal reproduction is printed out at the end of the debug log as a tried out repository, but the non working is not.

<details><summary>Logs</summary>

```
Using Kubernetes executor with image .../renovate/renovate:36.78.6 ...
...
DEBUG: Matched 1 file(s) for manager gradle: build.gradle (repository=renovate-36.78.6-gradle-repository-bug)
DEBUG: manager extract durations (ms) (repository=renovate-36.78.6-gradle-repository-bug)
       "managers": {"gradle": 23}
DEBUG: Found gradle package files (repository=renovate-36.78.6-gradle-repository-bug)
DEBUG: Found 1 package file(s) (repository=renovate-36.78.6-gradle-repository-bug)
 INFO: Dependency extraction complete (repository=renovate-36.78.6-gradle-repository-bug, baseBranch=main)
       "stats": {
         "managers": {"gradle": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: Looking up com.somegroup:some-dependency in repository https://some.working.repository.com/ (repository=renovate-36.78.6-gradle-repository-bug)
...
DEBUG: config.repoIsOnboarded=false (repository=renovate-36.78.6-gradle-repository-bug)
DEBUG: packageFiles with updates (repository=renovate-36.78.6-gradle-repository-bug, baseBranch=main)
       "config": {
         "gradle": [
           {
             "packageFile": "build.gradle",
             "datasource": "maven",
             "deps": [
               {
                 "depName": "com.somegroup:some-dependency",
                 "currentValue": "2.0.2",
                 "managerData": {
                   "fileReplacePosition": 93,
                   "packageFile": "build.gradle"
                 },
                 "fileReplacePosition": 93,
                 "datasource": "maven",
                 "registryUrls": ["https://some.working.repository.com/"],
                 "depType": "dependencies",
                 "updates": [],
                 "packageName": "com.somegroup:some-dependency",
                 "versioning": "gradle",
                 "warnings": [
                   {
                     "topic": "com.somegroup:some-dependency",
                     "message": "Failed to look up maven package com.somegroup:some-dependency"
                   }
                 ]
               }
             ]
           }
         ]
       }
...
```
</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
@Churro @zharinov Do you think we can support that syntax?

**Reply by rarkins** [reply]
If someone wants to add the logic to solve this, I think we can accept the PR. So this can be converted to a status:ready issue with help wanted, assuming @Churro doesn't want to solve it directly as he sometimes does :)

**Reply by zharinov** [reply]
Looks doable

**Comment by Churro** [comment]
While reimplementing the parser, I found several cases where matching `url` only would lead to collecting invalid repo URLs. The effect are significantly longer renovate runs as those will be used for all deps.

`name` is optional and there may be additional params within `{`, which makes matching fuzzy.

Support for this could be added in theory but since it's an edge case I'd suggest to rather specify / overwrite `registryUrls` in renovate.
